src/content/operations/nat-traversal/_index.en.md

```diff
@@ -0,0 +1,415 @@
+`PrivateIP` which is the IP assigned an interface on the gateway node where
+that `Endpoint` belongs, and `PublicIP` which is the src IP for the packets
+sent from the gateway to the Internet, currently we discover such IP via
+[ipify.org](https://ipify.org) but other mechanisms will be added in the future.
+
+### Reachability
+
+For two endpoints to connect to each other, at least one of them should be reachable
+either on the public or private IP address and firewall configuration should allow the
+tunnel encapsulation port.
+Using the `gateway.submariner.io/preferred-server=true` label only the labeled side
+needs to be reachable to other endpoints. For example:
+
+```bash
+kubectl label node $NODE gateway.submariner.io/preferred-server=true
+```
+
+Each gateway implements a UDP NATT discovery protocol where each gateway queries the
+gateways of the remote clusters on both the public & private IPs, figuring out the
+most suitable IP for the cable driver to use with preference on the private IP
+and the NAT characteristics of the path.
+
+This protocol is disabled by default, and should be enabled by assigning an UDP
+port to each gateway node on each cluster.
+
+```bash
+kubectl label node $NODE gateway.submariner.io/natt-discovery-port=4490
+```
+
+If the protocol has been enabled, and no reachability is possible between the
+endpoints the default settings will be used (which can be controlled with the
+NAT field of the operator or --natt parameter during subctl join), `--natt=false`
+will attempt the connection using the PrivateIP, `--natt=true` will attempt the
+connection using the PublicIP.
+
+### IP selection algorithm
+
+The following flow chart describes the IP selection algorithm:
+
+<!-- source: https://app.diagrams.net/#G1e_M84l8iQH6U4QqD3-Jw2ypIWiRfLQkk -->
+![Public/Private IP selection](/images/natt/public_private_selection_algorithm.svg)
+
+### Port selection
+
+If the gateways of your cluster don't have public floating or elastic IPs assigned
+to them then it's recommended to use a separate UDP port for every node marked
+as gateway in your cluster, this will allow eventual port mapping on a router if
+when communicating to clusters on remote sites with no direct routing.
+
+{{% notice note %}}
+If your cluster is behind a router which will NAT your traffic it's recommended to
+map the open ports into your gateway node private IPs, see the port mapping
+section. It could temporarily work without mapping, because most routers when
+performing NAT to the external network will not randomize or modify the source
+port of packets, but **this will happen as soon as two connections collide
+over the same source port**.
+{{% /notice %}}
+
+## UDP Dataplane protocol (IPSEC & wireguard)
+
+By default submariner uses the 4500 UDP port for the dataplane. The cluster-wide default
+can be changed with the `subctl join` `--nattport` flag.
+
+Although, it's more advisable to specify the port to be used per gateway node:
```

Comment:
```suggestion
By default submariner uses the 4500 UDP port for the dataplane. This can be changed cluster-wide via the `--nattport` flag on join although it's more advisable to specify the port to be used per gateway node:
```